src/live.rs

```diff
@@ -67,6 +69,23 @@ pub fn iso_ignition_embed(config: &IsoIgnitionEmbedConfig) -> Result<()> {
         }
     };
 
+    let mut holder: CopiedFileHolder;
+    let mut embed: EmbedArea;
+
+    if use_stdout {
+        holder = CopiedFileHolder {
+            file: OpenOptions::new()
+                .read(true)
+                .open(&config.input)
+                .chain_err(|| format!("opening {}", &config.input))?,
+            copied_path: None,
+            complete: false,
+        };
+    } else {
+        holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
+    }
+    embed = EmbedArea::for_file(&mut holder.file)?;
```

Comment:
This isn't really using `CopiedFileHolder` correctly; `file` is supposed to be the output file, not the input.  It happens to work with the code paths we have now, but it's confusing.

I spent some time trying to refactor this more cleanly (as you may also have done) and wasn't able to find a good solution within the existing architecture.  I ended up refactoring `CopiedFileHolder` into an enum in #478, which could have a `Stdout` variant added.

What do you think about merging this part as-is, and then I'll rebase #478 on top and refactor this part?

Reply:
> This isn't really using `CopiedFileHolder` correctly; `file` is supposed to be the output file, not the input. It happens to work with the code paths we have now, but it's confusing.

I agree :sunglasses: 

> I spent some time trying to refactor this more cleanly (as you may also have done) and wasn't able to find a good solution within the existing architecture. I ended up refactoring `CopiedFileHolder` into an enum in #478, which could have a `Stdout` variant added.

Played a little and it was outside what I was comfortable doing as a simple contributor.

> What do you think about merging this part as-is, and then I'll rebase #478 on top and refactor this part?

:+1: